packages/ui/src/components/ValidatorList/KeystoresDataGrid.tsx

```diff
@@ -61,6 +69,106 @@ export default function KeystoresDataGrid({
     openDashboardTab();
   }, [validatorSummaryURL]);
 
+  useState<CustomValidatorUpdateRequest>();
+  const [validatorsSubscriptionMap, setValidatorsData] =
+    useState<SmoothStatusByPubkey>({});
+  const [oracleCallError, setOracleCallError] = useState<string>();
+
+  // Check that Smooth API returns an expected response format
+  function isValidOracleResponse(response: SmoothValidatorByIndexApiResponse): boolean {
```

Comment:
You're right, it could be done as you say, but there is a little bit more to it.

To consider all possible edgecases, Oracle needs to store info about unsubscribed operators. It returns them with a "notsubscribed" status. These are different than "not found" operators (operators that never subbed or unsubbed, so untracked by the oracle). 

The endpoint is like this to clarify "notsubscribed" from "not found", but still it could be done as you say

Reply:
better to leave it like this to not change the oracle again